pipenv/utils/dependencies.py

```diff
@@ -298,6 +298,31 @@ def is_constraint(dep):
     return constraints
 
 
+def prepare_default_constraint_file(
+    project,
+    dir=None,
```

Comment:
We should avoid shadowing built in python function name `dir` -- I recommend naming it `directory`.

Reply:
I fixed it.